logger/logger.go

```diff
@@ -1,24 +1,81 @@
 package logger
 
 import (
+	"fmt"
 	"io"
 	"time"
 
+	"github.com/jsternberg/zap-logfmt"
+	isatty "github.com/mattn/go-isatty"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
 )
 
 func New(w io.Writer) *zap.Logger {
+	config := NewConfig()
+	l, _ := config.New(w)
+	return l
+}
+
+func (c *Config) New(defaultOutput io.Writer) (*zap.Logger, error) {
+	w := defaultOutput
+	format := c.Format
+	if format == "console" {
```

Comment:
this will work, but you could also switch this to a white list

```go
if format != "json" && format != "logfmt" && format != "auto" {
    return nil, fmt.Errorf("unknown logging format: %s", format)
}
```

Reply:
The error message for most log formats is handled by `newEncoder`. The reason why this is handled specially is because I didn't want to modify `newEncoder` to accept `auto`. Since `newEncoder` can't make the decision since it doesn't have access to the `io.Writer`, I wanted to make it explicit that `console` was specified. Since that means console is a valid input, I had to make it so a user-specified console would be intercepted and would pretend to be an invalid format.